### PR TITLE
Fixed #945 Selenide automatically takes screenshots on any test failure with junit5 runner

### LINKED ISSUE #945: Selenide should automatically take screenshot in case of AssertionError Junit(5)
## The problem
Selenide (ScreenShotterExtention) don`t take screenshots in Junit5 with [Assertions class](https://junit.org/junit5/docs/5.4.0/api/org/junit/jupiter/api/Assertions.html) 

## Details
Similar fixed issue in junit4 functionality:
#79 #95  

## Code To Reproduce Issue [ Good To Have ]
`
@Test
    @DisplayName("Junit assert without screenshot")
    void googleTest()  {
        open("https://duckduckgo.com/");
        assertEquals("asd", "zxc");
    }

    @Test
    @DisplayName("Selenide assert and screenshot")
    void googleTest2() {
        open("https://www.google.com/");
        $("#q").shouldHave(Condition.text("fail test"));
    }
`

### PR DESCRIPTION
**Pull request by Artem034**
- Selenide automatically takes screenshots on any test failure with junit5 runner
 - refactor for using TestWatcher class
 - add SelenideLogger for listeners.

## Proposed changes
#945 

## Checklist
- [x] Checkstyle and unit tests pass locally with my changes by running `gradle check chrome htmlunit` command
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: statics/src/main/java/com/codeborne/selenide/junit5/ScreenShooterExtension.java [hunk 1]
```diff
@@ -61,19 +64,32 @@ public void beforeAll(final ExtensionContext context) {
...
   @Override
-  public void afterAll(final ExtensionContext context) {
-    Screenshots.finishContext();
+  public void testAborted(ExtensionContext context, Throwable cause) {
+
+  }
+
+  @Override
+  public void testFailed(ExtensionContext context, Throwable cause) {
+    if (!(cause instanceof UIAssertionError)) {
```
Comment (rosolko) [inline statics/src/main/java/com/codeborne/selenide/junit5/ScreenShooterExtension.java]: Look like generally you change a logic. In previous implementation we collect expection only if it's instance of `UIAssertionError` but you change this logic. Why?

**Comment by rosolko** [comment]
Add tests to proof your changes working as exptected

File: statics/src/main/java/com/codeborne/selenide/junit5/ScreenShooterExtension.java (hunk 1 above)
Comment (Artem034) [reply statics/src/main/java/com/codeborne/selenide/junit5/ScreenShooterExtension.java]: For taking screens for junit assertions. All the details you can see here: #945, and proposed logic will be consistent with other test runners(Junit4, testNG)

**Comment by evgeny-tumashchick** [comment]
Hi, any plans to finish this pull request?
Selenide logger usage in ScreenShooterExtension extension will be a great help to Allure

**Comment by asolntsev** [comment]
Yes, I am going to review this PR.
Sorry for the delay.

**Comment by asolntsev** [comment]
@Artem034 Your fix is generally ok, but I see one problem: we take 2 screenshots.
1. This extensions takes a screenshot (and just logs it),
2. Then extension triggers Allure listener (via `SelenideLogger.commitStep`) which also takes a screenshot. 

It would be nice to avoid taking 2 screenshots.

**Comment by asolntsev** [comment]
Closing the issue because the same job was done by commit https://github.com/selenide/selenide/commit/5414bc743469d0624e6f5efc.

But we still have an issue with 2 screenshots. Registered an issue https://github.com/selenide/selenide/issues/1002.

**Pull request closed without merging**